STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/inc/az_ulib_registry_api.h

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+/**
+ * @file
+ *
+ * @brief   The Registry Editor.
+ *
+ * The Registry Editor is capable of storing key value pairs and stored into flash.
+ */
+
+#ifndef AZ_ULIB_REGISTRY_API_H
+#define AZ_ULIB_REGISTRY_API_H
+
+#include "az_ulib_result.h"
+#include "azure/az_core.h"
+
+#ifndef __cplusplus
+#include <stdint.h>
+#else
+#include <cstdint>
+#endif /* __cplusplus */
+
+#include "azure/core/_az_cfg_prefix.h"
+
+#define AZ_ULIB_CONFIG_MAX_REGISTRY_ENTRIES 10
+
+typedef enum
+{
+  REGISTRY_USED = 0x00,
+  REGISTRY_FREE = 0xFF
+} az_registry_entry_type;
+
+/**
+ * @brief   Structure for the registry control block
+ *
+ * Every entry into the registry of the device will have a registry control block that contains
+ * relevant information to restore the state of the registry after device power cycle.
+ *
+ */
+
+typedef struct
+{
+  /** The enum that declares whether a registry slot is occupied or not.*/
+  az_registry_entry_type used_flag;
+
+  /** The az_span that contains pointer to a location in flash storing the key char string and the
+   * string's size.*/
+  az_span key;
+
+  /** The az_span that contains pointer to a location in flash storing the valye char string and the
+   * string's size.*/
+  az_span value;
+} az_registry_block;
+
+/**
+ * @brief   This function gets the az_span value associated with the given az_span key from the
+ * registry.
+ *
+ * This function goes through the registry comparing keys until it finds one that matches with the
+ * input.
+ *
+ * @param[in]   key                  The az_span key to look for within the registry.
+ * @param[in]   value                The az_span value corresponding to the input key
+ *
+ * @pre         \p key                            shall not be `NULL`.
+ * @pre         \p sizeof(key)                    shall greater than zero.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                        If retrieving a value from the registry was
+ *                                            successful
+ *      @retval #AZ_ERROR_ITEM_NOT_FOUND      If there are no values that correspond to the
+ *                                            given key within the registry. *
+ *      @retval #AZ_ERROR_ARG                 If the parameters passed into`az_ulib_registry_get`
+ *                                            violate requirements for this function.
```

Comment:
We don't return AZ_ERROR_ARG for precondition errors anymore. Please remove this error.

Reply:
Same for the other APIs.